### PR TITLE
MINOR: Add ignorable field check to `toStruct`

### PR DESCRIPTION
**Pull request by hachikuji**
If a field is not marked as ignorable, we should raise an exception if it has been set to a non-default value. This check already exists in `Message.write`, so this patch adds it to `Message.toStruct`.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by cmccabe** [comment]
LGTM.  Thanks, @hachikuji

File: core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala [hunk 1]
```diff
@@ -143,13 +145,14 @@ class MetadataRequestTest extends BaseRequestTest {
     val response1 = sendMetadataRequest(new MetadataRequest.Builder(Seq(topic1, topic2).asJava, true, ApiKeys.METADATA.latestVersion).build())
     checkAutoCreatedTopic(topic1, topic2, response1)
 
-    // V3 doesn't support a configurable allowAutoTopicCreation, so the fact that we set it to `false` has no effect
-    val response2 = sendMetadataRequest(new MetadataRequest(requestData(List(topic2, topic3), false), 3.toShort))
-    checkAutoCreatedTopic(topic2, topic3, response2)
+    // V3 doesn't support a configurable allowAutoTopicCreation, so disabling auto-creation is not supported
+    intercept[UnsupportedVersionException] {
+      sendMetadataRequest(new MetadataRequest(requestData(List(topic2, topic3), false), 3.toShort))
+    }
```
Comment (ijuma) [inline core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala:157]: The original intent was for this to be ignorable. Have we decided that this is a bad idea?

Comment (hachikuji) [reply core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala:157]: I wasn't sure, so left it as currently defined. Note that the MetadataRequest builder pretty much already makes it non-ignorable.

**Review by omkreddy — APPROVED** [review]
@hachikuji Thanks for the PR. authorized operations related changes LGTM.

**Comment by hachikuji** [comment]
retest this please

File: core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala (hunk 1 above)
Comment (cmccabe) [reply core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala:157]: Hmm.  I don't think allowAutoTopicCreation should be ignorable.  It seems like a relatively important request option that really changes the meaning of the request.

**Comment by cmccabe** [comment]
LGTM.  Thanks, @hachikuji

File: core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala (hunk 1 above)
Comment (ijuma) [reply core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala:157]: Do we have a check for the case where `true` is used with older request types?

Comment (hachikuji) [reply core/src/test/scala/unit/kafka/server/MetadataRequestTest.scala:157]: Let me add something here.

**Pull request merged**